src/ucp/api/ucp.h

```diff
@@ -1878,6 +1981,85 @@ ucs_status_ptr_t ucp_tag_send_sync_nb(ucp_ep_h ep, const void *buffer, size_t co
 
 /**
  * @ingroup UCP_COMM
+ * @brief Non-blocking stream receive operation.
+ *
+ * This routine receives a data that is described by the local address @a
+ * buffer, size @a count, and @a datatype object on the endpoint @a ep. The
+ * routine is a non-blocking and therefore returns immediately. The receive
+ * operation is considered completed when the message is delivered to the @a
+ * buffer. In order to notify the application about completion of the receive
+ * operation the UCP library will invoke the call-back @a cb when the received
+ * data is in the receive buffer and ready for application access. If the
+ * receive operation cannot be stated the routine returns an error.
+ *
+ * @note This routine cannot return UCS_OK. It always returns a request
+ *       handle or an error.
```

Comment:
This note is no longer valid since, based on the discussion, some data could be immediately available.

However, I have a question about the number of elements returned.  Am I correct to assume that an immediate return could be for fewer elements than requested, but that a deferred receive only completes when all 'count' number of elements have been received?

Reply:
The recv can be completed immediately if any portion (not 0) of data is already arrived. We are going to add special flag like WAIT_ALL to complete the request only if [in] count of datatype entries are filled. In this case, [in] count always equal [out] count

Reply:
Sorry, answering on your question - yes, you are correct. The reasons are 1) to be aligned with socket semantic, 2) to be able to recv any amount of data from stream as soon as possible if needed.